Tickr/Tickr/StockManager.swift

```diff
@@ -0,0 +1,238 @@
+//
+//  StockManager.swift
+//  Tickr
+//
+//  Created by Stephen Payne on 4/9/16.
+//  Copyright © 2016 Stephen Payne. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+/*
+    A singleton class responsible for getting stock data 
+    from the API and communicating changes with any 
+    View Controllers that may be listening.
+*/
+class StockManager {
+    var isMakingRequest = false
+    var shouldCancelUpdate = false
+    var session = NSURLSession.self()
+    var stockArray = [Stock]()
+    lazy var notificationCenter: NSNotificationCenter = {
+        return NSNotificationCenter.defaultCenter()
+    }()
+    
+    //singleton initialization
+    class var sharedInstance : StockManager {
+        struct Static {
+            static let instance = StockManager()
+        }
+        return Static.instance
+    }
+    
+    
+    /*
+        This will take a search term and try to look up stock symbols or company names based on that term
+        - parameter term: A string to use as a search query
+     */
+    class func fetchStocksFromSearchTerm(term term: String, completion:(stockInfoArray: [StockSearchResult]) -> ()) {
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)) {
```

Comment:
you should keep a reference to the task, if it is running, then cancel it and restart another one, no need for the previous requests to finish